internal/datacoord/handler.go

```diff
@@ -333,9 +350,17 @@ func (h *ServerHandler) GetCollection(ctx context.Context, collectionID UniqueID
 }
 
 // CheckShouldDropChannel returns whether specified channel is marked to be removed
-func (h *ServerHandler) CheckShouldDropChannel(channel string) bool {
-	return h.s.meta.catalog.ShouldDropChannel(h.s.ctx, channel) ||
-		!h.s.meta.catalog.ChannelExists(h.s.ctx, channel)
+func (h *ServerHandler) CheckShouldDropChannel(channel string, collectionID UniqueID) bool {
+	if h.s.meta.catalog.ShouldDropChannel(h.s.ctx, channel) {
+		return true
+	}
+	// collectionID parse from channelName
+	has, err := h.HasCollection(h.s.ctx, collectionID)
+	if err != nil {
+		log.Info("datacoord ServerHandler CheckShouldDropChannel hasCollection failed", zap.Error(err))
+		return false
+	}
```

Comment:
maybe we should retry until get a non-error response and panic after max attempts?